include/seqan3/search/detail/search_configurator.hpp

```diff
@@ -47,12 +48,15 @@ class search_configurator
         using index_cursor_type = typename index_t::cursor_type;
         //!\brief The size type of the index.
         using index_size_type = typename index_t::size_type;
+        //!\brief The search traits.
+        using traits = search_traits<search_configuration_t>;
 
     public:
         //!\brief The result type depending on the output configuration.
-        using type = std::conditional_t<search_traits<search_configuration_t>::search_return_index_cursor,
-                                        search_result<size_t, index_cursor_type, empty_type, empty_type>,
-                                        search_result<size_t, empty_type, index_size_type, index_size_type>>;
+        using type = search_result<std::conditional_t<traits::output_query_id, size_t, empty_type>,
+                                   std::conditional_t<traits::output_index_cursor, index_cursor_type, empty_type>,
+                                   std::conditional_t<traits::output_reference_id, index_size_type, empty_type>,
+                                   std::conditional_t<traits::output_reference_begin_pos, index_size_type, empty_type>>;
```

Comment:
Can you make them type aliases before instead of having such a nested conditional   type definition?

Reply:
I think it looks quite clean. Anyway, I will change them but I think this is the kind of personal flavour that we regarded as rather "picky" no? :innocent: